src/utils/authentication.js

```diff
@@ -13,15 +13,29 @@ const signupValidation = [
     .withMessage('Your password must be at least eight charachters'),
 ];
 
-const generateToken = (user) => jwt.sign(
-  {
-    _id: user._id,
-    email: user.email,
-    username: user.username,
-    role: user.role,
-  },
-  process.env.JWT_SECRET,
-);
+const generateToken = user =>
+  jwt.sign(
+    {
+      _id: user._id,
+      email: user.email,
+      username: user.username,
+      role: user.role,
+    },
+    process.env.JWT_SECRET,
+    { expiresIn: 60 },
+  );
+
+const generateRefreshToken = user =>
+  jwt.sign(
+    {
+      _id: user._id,
+      email: user.email,
+      username: user.username,
+      role: user.role,
+    },
+    process.env.JWT_SECRET,
+    { expiresIn: '30d' },
```

Comment:
I tried to test refreshing logic so I set` { expiresIn: 2 }`but it did not work and gave me infinite error:
```
2021-07-21 13:25:37 error: Invalid token
POST /api/refreshToken 401 0.544 ms - 27
```
also FE did not handle it and did not log me out

Reply:
I checked every step in BE & FE by console logging and also checking in the actual app at the same time . I could get both token and refreshtoken in logs

Reply:
i also got the same result ://